discovery-provider/src/tasks/index_rewards_manager.py

```diff
@@ -0,0 +1,374 @@
+import concurrent.futures
+import logging
+import time
+from typing import Callable, List, TypedDict, Optional
+from sqlalchemy import desc
+from sqlalchemy.orm.session import Session
+from solana.rpc.api import Client
+
+import base58
+from src.models.models import User, UserChallenge
+from src.models import ChallengeDisbursement
+from src.tasks.celery_app import celery
+from src.utils.config import shared_config
+from src.utils.session_manager import SessionManager
+from src.utils.solana_indexing import (
+    ResultMeta,
+    TransactionMessage,
+    TransactionMessageInstruction,
+    get_sol_tx_info,
+    parse_instruction_data,
+    InstructionFormat,
+    SolanaInstructionType,
+    TransactionInfoResult,
+)
+
+logger = logging.getLogger(__name__)
+
+REWARDS_MANAGER_PROGRAM = shared_config["solana"]["rewards_manager_program_address"]
+MIN_SLOT = int(shared_config["solana"]["rewards_manager_min_slot"])
+
+# Maximum number of batches to process at once
+TX_SIGNATURES_MAX_BATCHES = 20
+
+# Last N entries present in tx_signatures array during processing
+TX_SIGNATURES_RESIZE_LENGTH = 10
+
+
+def check_valid_rewards_manager_program():
+    try:
+        base58.b58decode(REWARDS_MANAGER_PROGRAM)
+        return True
+    except ValueError:
+        logger.error(
+            f"index_rewards_manager.py"
+            f"Invalid Rewards Manager program ({REWARDS_MANAGER_PROGRAM}) configured, exiting."
+        )
+        return False
+
+
+is_valid_rewards_manager_program = check_valid_rewards_manager_program()
+
+
+rewards_manager_transfer_instr: List[InstructionFormat] = [
+    {"name": "amount", "type": SolanaInstructionType.u64},
+    {"name": "id", "type": SolanaInstructionType.string},
+    {"name": "eth_recipient", "type": SolanaInstructionType.EthereumAddress},
+]
+
+
+class RewardsManagerTransfer(TypedDict):
+    amount: int
+    id: str
+    eth_recipient: str
+
+
+def parse_transfer_instruction_data(data: str) -> RewardsManagerTransfer:
+    """Parse Transfer instruction data submitted to Audius Rewards Manager program
+
+    Instruction struct:
+    pub struct TransferArgs {
+        /// Amount to transfer
+        pub amount: u64,
+        /// ID generated on backend
+        pub id: String,
+        /// Recipient's Eth address
+        pub eth_recipient: EthereumAddress,
+    }
+
+    Decodes the data and parses each param into the correct type
+    """
+
+    return parse_instruction_data(data, rewards_manager_transfer_instr)
+
+
+def parse_transfer_instruction_id(transfer_id: str) -> List[str]:
+    """Parses the transfer instruction id into [challenge_id, specifier]
+    The id in the transfer instruction is formatted as "<CHALLENGE_ID>:<SPECIFIER>"
+    """
+    id_parts = transfer_id.split(":")
+    if len(id_parts) != 2:
+        raise Exception(
+            f"Unable to parse transfer instruction id into challenge_id and specifier {transfer_id}"
+        )
```

Comment:
If the transfer instruction is not formatted correctly, should we just skip it?
I wouldn't want that to stall indexing. Thoughts @piazzatron